lib/core/timer.cpp

```diff
@@ -14,4 +14,11 @@ Duration Timer::measure_execution_time(const std::function<void(void)> task)
   return after - before;
 }
 
+Duration Timer::measure_lapsed_time(const TimePoint previous_timepoint)
```

Comment:
naming style

Reply:
i think this whole paradigm is wrong, by the way

i fixed the timer class in the navigation benchmarker PR but never got around to finishing and merging that

maybe you can extract the stuff from there? it should give you the utility that you need

Reply:
I see what you've done in the benchmark pr but once the accelerometer setup is working, this will probably have to change to use the actual timepoints given with input data, making your other implementation of timer with the reset function not  suited to this. Think I'll change how this works to your implementation for now with a TODOLater for fixing this for final functionality.

Reply:
I've changed how this works after rethinking what information will actually be passed to the class and its methods. Timer utility wasn't there already but was probably quite a lot simpler than the previous implementation.